bee-storage/bee-storage/src/access/stream.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::backend::StorageBackend;
+
+use futures::stream::Stream;
+
+/// `AsStream<'a, K, V>` trait extends the `StorageBackend` with `stream` operation for the (key: K, value: V) pair;
+/// therefore, it should be explicitly implemented for the corresponding `StorageBackend`.
+#[async_trait::async_trait]
+pub trait AsStream<'a, K, V>: StorageBackend {
```

Comment:
If this was changed to `IntoStream` (in the style of `IntoIterator`) then it could consume `self` which makes it more generally applicable (it can still be implemented for all current cases by being implemented by a reference to them)